modules/math/src/geospatial/helpers/scale-to-geodetic-surface.js

```diff
@@ -40,29 +40,33 @@ export default function scaleToGeodeticSurface(cartesian, ellipsoid, result) {
   const squaredNorm = x2 + y2 + z2;
   const ratio = Math.sqrt(1.0 / squaredNorm);
 
+  if (!Number.isFinite(ratio)) {
+    return undefined;
+  }
+
   // As an initial approximation, assume that the radial intersection is the projection point.
-  const intersection = scaleToGeodeticSurfaceIntersection;
+  const intersection = new Vector3();
```

Comment:
We could pre-allocate this `Vector3` to avoid having to allocate an object here.